src/main/java/software/amazon/encryption/s3/internal/CipherSubscriber.java

```diff
@@ -125,7 +129,8 @@ public void onError(Throwable t) {
     public void onComplete() {
         // In rare cases, e.g. when the last part of a low-level MPU has 0 length,
         // onComplete will be called before onNext is called once.
-        if (contentRead.get() + tagLength <= contentLength) {
+        // tagLength should only be added on Encrypt
+        if (contentRead.get() + (isEncrypt ? tagLength : 0) >= contentLength) {
```

Comment:
The equality was flipped here -- was that intentional?
It seems like it was, since it now aligns with the other checks, and (maybe?) makes more sense conceptually

Reply:
Actually, the most sensible thing to do is to just call `finalBytes` from here regardless. In the vast majority of cases it'll already have been called, and simply return immediately; in the rare case(s) when it hasn't been we'll always want to send the final bytes.